docs/eBPFWeakReferences.md

```diff
@@ -0,0 +1,82 @@
+
+```
+| Event                              | Weak-ref object   | Referred-to object      |
+|                                    | ref count         | ref count               |
+|------------------------------------|-------------------|-------------------------|
+| Referred-to Obj creation           | 1                 | 1                       |
+|   (weak-obj is created alongside)  |                   |                         |
+|                                    |                   |                         |
+| 'borrow' weak-ref, (get ptr to     | increment         | unchanged               |
+|   weak-ref associated with         |                   |                         |
+|   referred-to obj)                 |                   |                         |
+|                                    |                   |                         |
+| try-acquire referred-to            | unchanged         | increment, iff non-zero |
+|   obj via weak-ref                 |                   |                         |
+|                                    |                   |                         |
+| release referred-to obj ref        | unchanged         | decrement               |
+|   (decrements ref count of         |                   |                         |
+|   referred-to obj)                 |                   |                         |
+|                                    |                   |                         |
+| weak-ref acquire ref               | increment         | unchanged               |
+|                                    |                   |                         |
+| weak-ref release ref               | decrement         | unchanged               |
+|                                    |                   |                         |
+| Referred-to obj destruction        | decrement         | decrement               |
+|                                    |                   |                         |
+| Weak-ref obj destruction           | decrement         | unchanged               |
+|                                    |                   |                         |
+```
+
+**Weak-ref management API:**
+
+The weak-ref implementation extends the base "ebpf object" management layer and is thus available for all primary object types.  The weak-ref management API are prototyped in the ```epbf_object.h``` header and provides the following calls:
+
+
+```ebpf_weak_reference_t* ebpf_object_weak_reference_get_reference(ebpf_core_object_t* object);```
+<br>
+The caller typically begins by calling this API to get a weak-ref pointer to the primary object they are interested in. This call increments the ref count for the weak-ref but does not modify the primary object's ref-count.  The caller now has a 'token' that they can now use to (conditionally) get the primary object pointer, either right away or at some later point in time.
```

Comment:
```suggestion
The caller typically begins by calling this API to get a weak-ref pointer to the primary object it is interested in. This call increments the ref count for the weak-ref but does not modify the primary object's ref-count.  The caller now has a 'token' that it can now use to (conditionally) get the primary object pointer, either right away or at some later point in time.
```

Reply:
fixed